spec/src/main/asciidoc/configprofile.asciidoc

```diff
@@ -0,0 +1,54 @@
+//
+// Copyright (c) 2020 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// Contributors:
+// Emily Jiang
+
+
+[[configprofile]]
+== Config Profile
+
+Config Profile indicates the project phrase, such as dev, testing, live, etc. 
+=== Specify Config Profile
+
+The config profile can be specified via a system property `mp.config.profile`. It can be set when starting your microservice, e.g.
+[source, text]
+----
+java -jar myapp.jar -Dmp.config.profile=testing
+----
+
+=== How Config Profile works
+
+In supporting the Config Profile, the config property names will need to start with the naming convention of `%configProfile%`, (e.g. `%live%vechile.name`).
```

Comment:
I thought about this. If we use %%, we can clearly demonstrate the profile. In your scenario, when you do the lookup, you will use `@ConfigProperty(name="config.key")`. Basically Quarkus use % and . to separate the profile name. I think starting and ending with % is better.

Reply:
No, we don't use `%` as a separator.  We prepend it on to the profile name and use that as a config segment.  This means that we only have one separator which is `.`.  Anyway I think MP ought to standardize working approaches rather than invent new ones.  Is anything else using this double `%` strategy?

Reply:
This is the first occurrence of this kind of naming convention as far as I know. Personally I feel %xxx% is standing out :o.

Reply:
Remember that our goal isn't to stand out. AFAIK our goal is to standardize existing best practices, and all existing implementations of this functionality use `%profilename.` as a prefix.

Reply:
I stand with @dmlloyd in this one. It's better to not deviate from working solutions

Reply:
Maybe I'm already biased, but %[profile].[key] seems more readable than %[profile]%key

Reply:
By the way, the spec has no obligation to adopt the existing implementation. It can come up with different alternatives or completely different designs to address the same problems. This is the practices we have been using. For this matter, since you three are from the same project, I think it is good idea to hear what others think, especially users.

Reply:
This discussion follows in the new comment thread: https://github.com/eclipse/microprofile-config/pull/554/files#r419090888